yarn-project/simulator/src/public/db.ts

```diff
@@ -92,6 +92,12 @@ export interface CommitmentsDB {
     secret: Fr,
   ): Promise<MessageLoadOracleInputs<typeof L1_TO_L2_MSG_TREE_HEIGHT>>;
 
+  /**
+   * @param leafIndex the leaf to look up
+   * @returns The l1 to l2 leaf value or undefined if not found.
+   */
+  getL1ToL2LeafValue(leafIndex: bigint): Promise<Fr | undefined>;
```

Comment:
Instead of introducing `getL1ToL2LeafValue`, IIUC I could change `getL1ToL2MembershipWitness` to take a "startIndex". Then pass the expected index, and check that the index matches after return. This would be slower when it's not found.